google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/DirectPathEnv.java

```diff
@@ -71,29 +77,50 @@ private DirectPathEnv(
     this.projectId = projectId;
     this.instanceId = instanceId;
     this.tableId = tableId;
-  }
 
-  @Override
-  void start() throws IOException {
+    // Build the data client settings
     BigtableDataSettings.Builder settingsBuilder =
         BigtableDataSettings.newBuilder().setProjectId(projectId).setInstanceId(instanceId);
 
     // Direct path test environment uses a different endpoint.
     settingsBuilder.stubSettings().setEndpoint(DIRECT_PATH_END_POINT);
 
-    // TODO(igorbernstein): move direct path conditional logic to gax
-    // Direct path environment can be accessed via CFE or direct path. When using direct path,
-    // disable connection pooling.
+    // TODO: Remove this after https://github.com/googleapis/gax-java/pull/707 is merged
     if (directPathEnabled) {
       settingsBuilder
           .stubSettings()
           .setTransportChannelProvider(
               EnhancedBigtableStubSettings.defaultGrpcTransportProviderBuilder()
-                  .setPoolSize(1)
+                  .setChannelConfigurator(
+                      new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+                        @Override
+                        public ManagedChannelBuilder apply(ManagedChannelBuilder builder) {
+                          // Configure pick_first strategy to force a single subchannel per
+                          // ManagedChannel
+                          builder.defaultServiceConfig(
+                              ImmutableMap.of(
```

Comment:
this is hard to follow, consider using more local variables

Reply:
Done